[Feature]: add option to export to opensensemap compatible csv

mashazyu opened the pull request:
Closes #44 

Add option to export track data to the .csv format ready for upload to openSenseMap.

### Detailed Changes

*   Add prompt to select desired .csv format
*   Add CircularListTile widget to display selectable option in a list
*   Tests

### Testing

- User should be able to export track data to the ,csv file and import it in the openSenseMap portal.
- Regular .csv export should work as before.

### Checklist before requesting a review

[ ] I have performed a self-review of my code
[ ] I have added or updated tests
[ ] This change gracefully effects existing users
[ ] I have added or updated relevant documentation

### Additional Information

mashazyu commented on lib/constants.dart:
```diff
@@ -3,6 +3,14 @@ const retryPeriod = 2; // in minutes
 const premanentConnectivityFalurePeriod = 20; // in minutes
 const defaultTimeout = 30; // in seconds
 
+// Constants for styling
```
**suggestion:** I started moving values used in the design to the constants file.

felixerdy replied:
Would it be better to put it in the `lib/theme.dart` file?

mashazyu commented on lib/services/isar_service.dart:
```diff
@@ -20,14 +22,39 @@ class IsarService {
   final GeolocationService geolocationService = GeolocationService();
   final SensorService sensorService = SensorService();
 
+  Future<TrackData> _getTrackOrThrow(int trackId) async {
```
**suggestion:** I refactored IsarService to make it 
- more DRY
- easier to test

mashazyu commented on lib/ui/screens/home_screen.dart:
```diff
@@ -88,13 +91,9 @@ class _SenseBoxSelectionButton extends StatelessWidget {
-            width: 1.5,
+                ? colorScheme.error
+                : colorScheme.tertiary,
+            width: borderWidth,
```
**suggestion:** Minor refactoring to reuse exported style value.

mashazyu commented on lib/ui/widgets/common/button_with_loader.dart:
```diff
@@ -5,35 +5,40 @@ class ButtonWithLoader extends StatelessWidget {
     required this.onPressed,
     required this.text,
-    required this.width,
+    this.width, // Optional
```
**suggestion:** Made width option to easier reuse this widget.

mashazyu commented on lib/ui/widgets/common/error_dialog.dart:
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
```
**suggestion:** Extracted error dialog to reuse it later on.

mashazyu commented on lib/utils/isar_utils.dart:
```diff
@@ -0,0 +1,94 @@
+import 'dart:convert';
```
**suggestion:** Here I exported csv building logic to a separate file.

felixerdy commented on lib/l10n/app_de.arb:
```diff
@@ -149,5 +150,8 @@
-    "errorNoSenseBoxSelected": "Bitte loggen Sie sich in Ihr openSenseMap-Konto ein und wählen Sie eine Box aus, um Sensordaten in die Cloud hochzuladen."
+    "errorNoSenseBoxSelected": "Bitte loggen Sie sich in Ihr openSenseMap-Konto ein und wählen Sie eine Box aus, um Sensordaten in die Cloud hochzuladen.",
+    "selectCsvFormat": "CSV-Format auswählen",
+    "regularCsv": "Reguläres CSV",
```
I don't like the "Regulär". What about:
```
• Standard (CSV)
• openSenseMap (CSV)
```

mashazyu replied:
Much better

felixerdy commented on lib/ui/widgets/common/circular_list_tile.dart:
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:sensebox_bike/constants.dart';
+
+class CircularListTile extends StatelessWidget {
```
That's all right for me. Maybe just add a small comment at the top of the file

felixerdy reviewed (changes requested):
Great work. Only had a few comments

mashazyu commented:
@felixerdy 

Thank you for the review! I addressed your comments and renamed CircularListTile to SelectableListTile.
As we discussed confirmed that on main app produces the same file for the standard .csv use case for the following tracks:
- regular track,
- track without sensor data (gps only),
- track with 1 location
(I want to disallow export for the empty tracks, so I didn't use this scenario).

For the oSM csv I tested that the following tracks can be imported:
- without sensor data (only gps)
- file with partial sensor data (only part of the track had sensor data)

If there is no sensor/track data, export option is disabled.

Could you please take a look at this PR once again?
